src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -682,6 +632,177 @@ class EraseGather : public ngraph::pass::MatcherPass {
 };
 }  // namespace fuse
 
+namespace back_prop {
+class Binary : public ngraph::pass::MatcherPass {
+public:
+    Binary() {
+        MATCHER_SCOPE(Binary);
+        auto pattern_root = pattern::wrap_type<op::util::BinaryElementwiseArithmetic, opset8::FakeQuantize>();
+
+        auto callback = [=](pattern::Matcher& m) {
+            const auto& root = m.get_match_root();
+            const auto& output = root->output(0);
+
+            auto inputs = output.get_target_inputs();
+            std::vector<ric_attr::Attribute> attrs;
+            for (const auto& input : inputs) {
+                if (ric_attr::has(input)) {
+                    attrs.push_back(ric_attr::get(input).propagate());
+                } else {
+                    return false;
+                }
+            }
+
+            // Check that all RIC attrs can be merged and then merge them
+            auto ric = attrs[0];
+            auto rank = root->get_output_partial_shape(0).rank();
+            if (rank.is_dynamic())
+                return false;
```

Comment:
Please use pattern predicate to check this during matching.

Reply:
could you resolve this one? I guess we should use something like this `pattern::has_static_shape()`